### PR TITLE
MBS-9515: Block anchors in new Wikipedia URLs

### PR DESCRIPTION
**Pull request by yvanzo**
## [MBS-9515](https://tickets.metabrainz.org/browse/MBS-9515): Block adding Wikipedia relationships to anchors

### CODE REVIEW

File: root/static/scripts/edit/externalLinks.js [hunk 1]
```diff
@@ -196,6 +196,11 @@ class ExternalLinksEditor extends React.Component<LinksEditorProps, LinksEditorS
               error = l('This relationship type is deprecated and should not be used.');
             } else if ((!isPositiveInteger(link.relationship) || (oldLink && link.url !== oldLink.url)) && checker && !checker(link.url)) {
               error = l('This URL is not allowed for the selected link type, or is incorrectly formatted.');
+            } else if ((!isPositiveInteger(link.relationship) || (oldLink && link.url !== oldLink.url)) && /^(https?:\/\/)?([^.\/]+\.)?wikipedia\.org\/.*#/.test(link.url)) {
+              // Kludge for MBS-9515 to be replaced with the more general MBS-9516
+              error = l('Anchor (#Example_section) Wikipedia links are not allowed, see the {url|guidelines for URL}.', {
+                url: { href: '/doc/Style/Relationships/URLs', target: '_blank' }
```
Comment (reosarevok) [inline root/static/scripts/edit/externalLinks.js]: This should probably link to something like https://musicbrainz.org/relationship/29651736-fa6d-48e4-aadc-a557c6add1cb 

It's kind of annoying that we don't have a central "Wikipedia page" rel to link to, but only one per entity type :/ But linking to a specific one, like artist, might be better than nothing.

File: root/static/scripts/edit/externalLinks.js [hunk 2]
```diff
@@ -196,6 +196,12 @@ class ExternalLinksEditor extends React.Component<LinksEditorProps, LinksEditorS
               error = l('This relationship type is deprecated and should not be used.');
             } else if ((!isPositiveInteger(link.relationship) || (oldLink && link.url !== oldLink.url)) && checker && !checker(link.url)) {
               error = l('This URL is not allowed for the selected link type, or is incorrectly formatted.');
+            } else if ((!isPositiveInteger(link.relationship) || (oldLink && link.url !== oldLink.url)) && /^(https?:\/\/)?([^.\/]+\.)?wikipedia\.org\/.*#/.test(link.url)) {
+              // Kludge for MBS-9515 to be replaced with the more general MBS-9516
+              error = l('Anchors Wikipedia links are not allowed, please remove {fragment} if still appropriate, see the {url|guidelines}.', {
```
Comment (mwiencek) [inline root/static/scripts/edit/externalLinks.js]: Will some users know what an anchor is? I guess it's fine, but I wonder if something like "Wikipedia links to specific sections of the page are not allowed, ..." would make more sense.

The comma splice at the end sounds slightly unnatural so I'd replace "appropriate, see" with "appropriate. See" too.

File: root/static/scripts/edit/externalLinks.js [hunk 3]
```diff
@@ -196,6 +196,12 @@ class ExternalLinksEditor extends React.Component<LinksEditorProps, LinksEditorS
...
+                fragment: '<i>' + _.escape(link.url.replace(/^(?:https?:\/\/)?(?:[^.\/]+\.)?wikipedia\.org\/[^#]*#(.*)$/, '#$1')) + '</i>',
```
Comment (mwiencek) [inline root/static/scripts/edit/externalLinks.js]: It's much safer to pass `__react: true` here and pass in a React element directly, like `fragment: <i>{link.url.replace(...)}</li>`. Then you can revert the `dangerouslySetInnerHTML` below.

Comment (mwiencek) [reply root/static/scripts/edit/externalLinks.js]: Note that for the URL syntax you don't need to pass in an `<a>`, the i18n module will do that for you with `__react: true`. So the `url: {...}` part can stay as it is.

File: root/static/scripts/edit/externalLinks.js (hunk 1 above)
Comment (yvanzo) [reply root/static/scripts/edit/externalLinks.js]: https://github.com/metabrainz/musicbrainz-server/pull/562/commits/46feadc2646625daf81abd03ce12adc529f33f23 links to the specific relationship page for the current URL.
Besides, I reported the lack of central page for similar relationship types in [MBS-9517](https://tickets.metabrainz.org/browse/MBS-9517).

File: root/static/scripts/edit/externalLinks.js (hunk 2 above)
Comment (yvanzo) [reply root/static/scripts/edit/externalLinks.js]: Almost applied in https://github.com/metabrainz/musicbrainz-server/pull/562/commits/46feadc2646625daf81abd03ce12adc529f33f23

File: root/static/scripts/edit/externalLinks.js (hunk 3 above)
Comment (yvanzo) [reply root/static/scripts/edit/externalLinks.js]: Applied in https://github.com/metabrainz/musicbrainz-server/pull/562/commits/46feadc2646625daf81abd03ce12adc529f33f23, with extra style in https://github.com/metabrainz/musicbrainz-server/pull/562/commits/4007afdc64f65f392f1a769eaca344d18773fe37 that wraps the quoted URL fragment in order to break it if longer than line.

**Comment by yvanzo** [comment]
Selenium tests seem to deadlock [build #631](https://ci.metabrainz.org/job/musicbrainz-server/631/).

**Pull request merged**